src/index.ts

```diff
@@ -104,18 +104,40 @@ export interface Options {
 export const isJSON = (input: string) => input.match(/^[\[\{\"\}].*[\]\}\"]$/);
 
 function convertUnconventionalData(data: unknown) {
+  if (!isObject(data)) {
+    return data;
+  }
+
+  let result: any = data;
+  let wasMutated = false;
+
   // `Event` has a weird structure, for details see `extractEventHiddenProperties` doc
   // Plus we need to check if running in a browser to ensure `Event` exist and
   // is really the dom Event class.
   if (isRunningInBrowser && data instanceof Event) {
-    return extractEventHiddenProperties(data);
+    result = extractEventHiddenProperties(result);
+    wasMutated = true;
   }
 
-  return data;
+  result = Object.keys(result).reduce((acc, key) => {
+    try {
+      // Try accessing a property to test if we are allowed to do so
+      // eslint-disable-next-line no-unused-expressions
+      result[key]?.toJSON;
```

Comment:
I think a good way of testing this could be through a proxy. That seems to behave identical to what we see with the cross origin iframe.

Something like this.

```ts
test('filter for forbidden objects', () => {
  const thrower = {
    a: 'foo',
    b: new Proxy({}, {
      get() {
        throw new Error('b is not allowed!');
      }
    }),
  };
  const stringified = stringify(thrower);
  const parsed = parse(stringified);

  expect(parsed).toEqual({ a: 'foo' });
});
```